RHDEVDOCS-3283 Document advanced prunning options (#40748)
openshift/openshift-docs: modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc

// This module is included in the following assembly:
//
// cicd/pipelines/automatic-pruning-taskrun-pipelinerun.adoc

[id="annotations-for-automatic-pruning-taskruns-pipelineruns_{context}"]
= Annotations for automatically pruning task runs and pipeline runs 

To automatically prune task runs and pipeline runs in a namespace, you can set the following annotations:

* `operator.tekton.dev/prune.schedule`: If the value of this annotation is different from the value specified in the `TektonConfig` custom resource definition, a new cron job in that namespace is created.

> * `operator.tekton.dev/prune.skip`: When set to `true`, the namespace for which it is configured will not be pruned.

Reviewer: ```suggestion
* `operator.tekton.dev/prune.skip`: When set to `true`, the namespace for which it is configured is not pruned.
```

> * `operator.tekton.dev/prune.resources`: This annotation accepts a comma-separated list of resources. To prune a single resource such as a pipeline run, set this annotation to `"pipelinerun"`. To prune multiple resources, such as task run and pipeline run, set this annotation to `"taskrun, pipelinerun"`.

Reviewer: You want the user to enter the annotation between double-quotes, correct?

Author: Yes, it's treated as a string as a whole.

* `operator.tekton.dev/prune.keep`: Use this annotation to retain a resource without pruning.

* `operator.tekton.dev/prune.keep-since`: Use this annotation to retain resources based on their age. The value for this annotation must be equal to the age of the resource in minutes. For example, to retain resources which were created not more than five days ago, set `keep-since` to `7200`.
+
[NOTE]
====
The `keep` and `keep-since` annotations are mutually exclusive. For any resource, you must configure only one of them.
====

* `operator.tekton.dev/prune.strategy`: Set the value of this annotation to either `keep` or `keep-since`.

> As an example, consider the following annotations that retain all task runs and pipeline runs created in the last 5 days, and deletes the older resources: 

Reviewer: ```suggestion
For example, consider the following annotations that retain all task runs and pipeline runs created in the last 5 days, and deletes the older resources: 
```

.Example of auto-pruning annotations
[source,yaml]
----
...

>   annotations: 

Reviewer 2: :+1:
